contracts/DeploymentValidator.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "./libraries/Authorizable.sol";
+import "./interfaces/IDeploymentValidator.sol";
+
+contract DeploymentValidator is IDeploymentValidator, Authorizable {
+    // a mapping of wrapped position contracts deployed by Element
+    mapping(address => bool) public wrappedPositions;
+    // a mapping of pool contracts deployed by Element
+    mapping(address => bool) public pools;
+    // a mapping of wrapped position + pool pairs that are deployed by Element
+    // we keccak256 hash these tuples together to serve as the mapping keys
+    mapping(bytes32 => bool) public pairs;
+
+    /// @notice Constructs this contract and stores needed data
+    /// @param _owner The contract owner authorized to validate addresses
+    constructor(address _owner) {
+        // authorize the owner address to be able to execute the validations
+        _authorize(_owner);
+    }
+
+    /// @notice adds a wrapped position address to the mapping
+    /// @param wrappedPosition The wrapped position contract address
+    function validateWPAddress(address wrappedPosition)
+        external
+        override
+        onlyAuthorized
+    {
+        // add address to mapping to indicating it was deployed by Element
+        wrappedPositions[wrappedPosition] = true;
+    }
+
+    /// @notice adds a wrapped position address to the mapping
+    /// @param pool the pool contract address
+    function validatePoolAddress(address pool)
+        external
+        override
+        onlyAuthorized
+    {
+        // add address to mapping to indicating it was deployed by Element
+        pools[pool] = true;
+    }
+
+    /// @notice adds a wrapped position + pool pair of addresses to mapping
+    /// @param wrappedPosition the wrapped position contract address
+    /// @param pool the pool contract address
+    function validateAddresses(address wrappedPosition, address pool)
```

Comment:
For this, could we have it also approve both of the other mappings so we only need one onchain call?

Reply:
good idea, added that in!